FEXCore/Source/Utils/FileLoading.cpp

```diff
@@ -24,24 +24,43 @@ static bool LoadFileImpl(T &Data, const fextl::string &Filepath, size_t FixedSiz
   size_t FileSize{};
   if (FixedSize == 0) {
     struct stat buf;
-    if (fstat(FD, &buf) != 0) {
-      close(FD);
-      return false;
+    if (fstat(FD, &buf) == 0) {
+      FileSize = buf.st_size;
     }
-
-    FileSize = buf.st_size;
   }
   else {
     FileSize = FixedSize;
   }
 
   ssize_t Read = -1;
-  if (FileSize > 0) {
-    Data.resize(FileSize);
-    Read = pread(FD, &Data.at(0), FileSize, 0);
+  bool LoadedFile{};
+  if (FileSize) {
+    // Filesize is known upfront.
+    if (FileSize > 0) {
+      Data.resize(FileSize);
+      Read = pread(FD, &Data.at(0), FileSize, 0);
+    }
+
+    LoadedFile = Read == FileSize;
+  }
+  else {
+    // Filesize is unknown, try reading.
```

Comment:
```suggestion
    // File size is unknown. Try reading in chunks instead
```

Reply:
In light of https://github.com/FEX-Emu/FEX/pull/3243/files#r1381421301 , this is worth extra emphasis:

```suggestion
    // The file is either empty or its size is unknown (e.g. procfs data).
    // Try reading in chunks instead
```